tests/main.test.ts

```diff
@@ -1021,76 +1021,93 @@ describe(`handle integrity checks`, () => {
       });
     });
   });
-  it(`should return an error when signature does not match with a tag`, async () => {
+  it(`fails on invalid signature for version range`, async () => {
     process.env.TEST_INTEGRITY = `invalid_signature`; // See `_registryServer.mjs`
 
     await xfs.mktempPromise(async cwd => {
       await expect(runCli(cwd, [`pnpm@1.x`, `--version`], true)).resolves.toMatchObject({
         exitCode: 1,
-        stderr: /Signature does not match/,
+        stderr: expect.stringMatching(/Error: Signature does not match/),
         stdout: ``,
       });
+    });
+  });
+
+  // FIXME: the test registry does not implement tag fetching yet
+  it.fails(`fails on invalid signature for tag`, async () => {
+    process.env.TEST_INTEGRITY = `invalid_signature`; // See `_registryServer.mjs`
+
+    await xfs.mktempPromise(async cwd => {
       await expect(runCli(cwd, [`yarn@stable`, `--version`], true)).resolves.toMatchObject({
         exitCode: 1,
-        stderr: /Signature does not match/,
+        stderr: expect.stringMatching(/Signature does not match/),
         stdout: ``,
       });
     });
   });
-  it(`should return an error when hash does not match without a tag`, async () => {
+
+  // FIXME: Corepack determines the latest version by asking npm instead of the
+  // configured registry. Consequently the test registry does not have the
+  // requested version.
```

Comment:
I don't think that's true, the tests on CI do not have access to the network, it would fail if it was getting the info from the npm registry

Reply:
IIUC Corepack is trying to download the version defined in `config.json`, because we set `COREPACK_DEFAULT_TO_LATEST` to `0` on line 16. The issue is that that version doesn't exist on the custom registry

Reply:
I've opened https://github.com/nodejs/corepack/pull/649 to fix the test instead